snap/hooks/install

```diff
@@ -1,28 +1,43 @@
-#!/bin/bash
+#!/usr/bin/env bash
+# Installation hook for charmed-mongodb snap
 
 set -eux
 
+export CONF="${SNAP_DATA}"/etc/mongod
+export PBM_CONF="${SNAP_DATA}"/etc/pbm
+export DATA="${SNAP_COMMON}"/var/lib/mongodb
+export LOGS="${SNAP_COMMON}"/var/log/mongodb
+export MONGO_CONFIG_FILE="${CONF}"/mongod.conf
+
 # Create the necessary parent directories 
-mkdir -p "${SNAP_DATA}/etc/pbm/"
-mkdir -p "${SNAP_DATA}/etc/mongod/"
-mkdir -p "${SNAP_COMMON}/var/lib/mongodb/"
-mkdir -p "${SNAP_COMMON}/var/log/mongodb/"
+mkdir -p $DATA
+mkdir -p $LOGS
+mkdir -p $PBM_CONF
+mkdir -p $CONF
+
+if [ stat -c '%u' "$DATA" == 0 -o ! -d "$DATA" ]; then
+    chmod -R 770 "${SNAP_COMMON}"
+    chmod 750 "${DATA}"
+fi
+
+if [ stat -c '%u' "$LOGS" == 0 -o ! -d "$LOGS" ]; then
+    chmod -R 770 "${SNAP_COMMON}"
+    chmod g+s "$LOGS"/*
+fi
```

Comment:
mind adding comments for these staements 🙏 

noob question, does it hurt to do ` chmod -R 770 ` /  `chmod g+s ` /  `chmod 750 ` more than once? if not can we pull out of if statement?

Reply:
I tried and we cannot pull it out of the statement because the `chmod` command fails if out of the statement.
I'm checking for the two other commands.